include/gmsh.tcc

```diff
@@ -194,12 +194,111 @@ void GMSH<Tdim, Tvertices>::store_element_vertices() {
       << "The coordinates for vertices of each element have been stored\n";
 }
 
-//! Compute material points based on the centroid
+//! Compute material points for 2D quadrilateral
 //! \tparam Tdim Dimension
 //! \tparam Tvertices Number of vertices in element
-//! \param[in] ngauss_points Number of gauss points per coordinate
-template <unsigned Tdim, unsigned Tvertices>
-void GMSH<Tdim, Tvertices>::compute_material_points(unsigned ngauss_points) {
+//! \param[in] ngauss_points Number of gauss points per coordinate, element_type GMSH element type
+template <>
+inline void GMSH<2, 4>::compute_material_points(unsigned ngauss_points, unsigned element_type) {
+
+  const unsigned Tdim = 2;
+  const unsigned Tvertices = 4;
+
+  //! Storing ngauss_points to member variable and get constants from namespace
+  ngauss_points_ = ngauss_points;
```

Comment:
Why do we need this member variable `ngauss_points_`?

Reply:
now that we have `ngauss_points`. I can remove it